pilot/pkg/model/context.go

```diff
@@ -369,6 +369,12 @@ type Proxy struct {
 	// WatchedResources contains the list of watched resources for the proxy, keyed by the DiscoveryRequest TypeUrl.
 	WatchedResources map[string]*WatchedResource
 
+	// OrderedWatchedResources contains the watched resources for the proxy, ordered in accordance with known push order.
+	OrderedWatchedResources []*WatchedResource
+
+	// KnownTypeUrls contains the set of known type URLs for the proxy.
+	KnownTypeUrls sets.String
```

Comment:
We donot want to store all the results that can be used.

Reply:
What do you mean? This is just a set of type urls - which would be a maximum of 5/6 in this list